src/Symfony/Component/Validator/ConstraintValidator.php

```diff
@@ -105,6 +111,38 @@ protected function formatValue($value, $format = 0)
             return $value->format('Y-m-d H:i:s');
         }
 
+        if (($format & self::PRETTY_DATE_INTERVAL) && $value instanceof \DateInterval) {
+            $formattedValueParts = [];
+            foreach ([
+                'y' => 'year',
+                'm' => 'month',
+                'd' => 'day',
+                'h' => 'hour',
+                'i' => 'minute',
+                's' => 'second',
+                'f' => 'microsecond',
+            ] as $p => $label) {
+                if (!($formattedValuePart = $value->format('%'.$p))) {
+                    continue;
+                }
+
+                $formattedValuePart .= ' '.$label;
+                if ($formattedValuePart > 1) {
+                    $formattedValuePart .= 's';
+                }
+
+                $formattedValueParts[] = $formattedValuePart;
+            }
+
+            if (!$formattedValueParts) {
+                return '0';
+            }
+
+            $lastFormattedValuePart = array_pop($formattedValueParts);
+
+            return $value->format('%r').(!$formattedValueParts ? $lastFormattedValuePart : implode(', ', $formattedValueParts).' and '.$lastFormattedValuePart);
```

Comment:
As I see `!$formattedValueParts` will never be `true`. I think that code can be simplified.

Reply:
It can be true because it is passed to `array_pop` by reference, so it can become empty.

Reply:
Formatting with `,` as a separator looks quite arbitrary to me.

Reply:
The formatted value is in English. Which separator should I use?